activation/e2e/builds_atx_v2_test.go

```diff
@@ -211,6 +216,6 @@ func TestBuilder_SwitchesToBuildV2(t *testing.T) {
 	tab.Register(sig)
 
 	require.NoError(t, tab.StartSmeshing(coinbase))
-	require.Eventually(t, ctrl.Satisfied, epoch*4, time.Millisecond*100)
+	require.Eventually(t, func() bool { expectedAtxs.Wait(); return true }, epoch*4, time.Millisecond*100)
```

Comment:
This is a blocking call

```suggestion
	expectedAtxs.Wait()
```

Reply:
The `require.Eventually` runs it on a goroutine and times out after `epoch *4`. Calling `expectedAtxs.Wait()` directly would block the test.

Reply:
But then instead of a waitgroup I would just add a channel `publishCalled` to the last expected gomock call and do this:

```suggestion
	select {
	case <-time.After(epoch*4):
		t.Fail("timed out waiting for ATX publications")
	case <-publishCalled:
	}

Reply:
Why? That's far more complicated and less self-explanatory.

Reply:
Your explanation isn't correct - `Eventually` will start a new go routine every 100 milliseconds, independent of if the previous one completed; in total 40 if the test times out.

The `select` method doesn't start any additional go routines.

Reply:
Changed to atomic counter